lib/xcode.js

```diff
@@ -224,9 +228,23 @@ async function getMaxIOSSDKWithoutRetry (timeout = XCRUN_TIMEOUT) {
   return sdkVersion;
 }
 
+async function getMaxIOSSDKFromXcodeVersion (timeout = XCRUN_TIMEOUT) {
+  const version = await getVersion(true, DEFAULT_NUMBER_OF_RETRIES, timeout);
+  // as of now, the iOS version associated with an Xcode version is
+  // just the Xcode version + 2
+  return `${version.major + 2}.${version.minor}`;
+
+}
+
 const getMaxIOSSDK = _.memoize(
   function getMaxIOSSDK (retries = DEFAULT_NUMBER_OF_RETRIES, timeout = XCRUN_TIMEOUT) {
-    return retry(retries, getMaxIOSSDKWithoutRetry, timeout);
+    try {
+      return retry(retries, getMaxIOSSDKWithoutRetry, timeout);
```

Comment:
why no await?

Reply:
I'm not sure. It wasn't an `async` function before, and I figured there must be a reason for that.

Reply:
None of the memoized functions are `async`, at least in this package.